src/app/clusters/webrtc-transport-provider-server/webrtc-transport-provider-server.h

```diff
@@ -0,0 +1,225 @@
+     *     negotiation. The specification states that the provider may be in a standby mode
+     *     (low power), thus a "deferred offer" scenario might occur if the device needs time
+     *     to fully power resources needed for streaming.
+     *
+     * @param[in]  args
+     *     Structure containing all input arguments for the command.
+     *
+     * @param[out] outSession
+     *     On success (`CHIP_NO_ERROR`), must be populated with the new session's details.
+     *
+     * @param[out] outDeferredOffer
+     *     On success (`CHIP_NO_ERROR`), set to `true` if the device is in standby
+     *     or needs to defer generating an Offer; otherwise, `false`.
+     *     If an error is returned, this value is not relevant.
+     *
+     * @return CHIP_ERROR
+     *     - CHIP_NO_ERROR if a session was successfully created or prepared for negotiation.
+     *     - CHIP_ERROR_NO_MEMORY if the device cannot allocate a new session.
+     *     - Appropriate error otherwise.
+     *
+     * The combination of `outDeferredOffer == true` with an error return code is invalid
+     * (the device cannot be "deferring" if it is failing the command).
+     */
+    virtual CHIP_ERROR HandleSolicitOffer(const OfferRequestArgs & args, WebRTCSessionTypeStruct & outSession,
+                                          bool & outDeferredOffer) = 0;
+
+    /**
+     * @brief
+     *   Handles the ProvideOffer command received by the server.
+     *
+     * @param[in]  args
+     *   Contains all input arguments for the command, including the SDP Offer, session usage, etc.
+     *
+     * @param[out] outSession
+     *   Must be populated with the final session details (session ID, stream IDs, etc.) when this
+     *   method returns `CHIP_NO_ERROR`. If an error is returned, `outSession` is left unmodified or
+     *   set to an invalid state.
+     *
+     * @return
+     *   - CHIP_NO_ERROR if the request succeeds and `outSession` is populated
+     *   - CHIP_ERROR_NO_MEMORY if the device cannot allocate a new session.
+     *   - Appropriate error otherwise.
+     */
+    virtual CHIP_ERROR HandleProvideOffer(const ProvideOfferRequestArgs & args, WebRTCSessionTypeStruct & outSession) = 0;
+
+    /**
+     * @brief
+     *   Called when the server receives the ProvideAnswer command.
+     *
+     *   In the standard WebRTC Offer/Answer flow, once the provider sends an Offer (or
+     *   the requestor has re-offered), the other side will respond with an SDP Answer.
+     *   This method is invoked to store and process that Answer.
+     *
+     * @param[in] sessionId  The identifier of the current session.
+     * @param[in] sdpAnswer  The SDP Answer string.
+     *
+     * @return CHIP_ERROR
+     *   - CHIP_NO_ERROR if the Answer is accepted
+     *   - Appropriate error if invalid or unexpected
+     */
+    virtual CHIP_ERROR HandleProvideAnswer(uint16_t sessionId, const std::string & sdpAnswer) = 0;
```

Comment:
It's odd to have both "Id" and "ID" in this API, depending on where.  Pick one?

Reply:
The confusion comes from the generated object which use both ID and Id, for server and delegate implementation, I choose all use Id

Reply:
```    
uint16_t GenerateSessionID();
```
?

Reply:
More generall, generated bits have ID because the spec has ID...  Whether you match that in your other code is up to you, I guess, but it does read weirdly.

Reply:
most of generated bits use 'Id', such as Attributes::CurrentSessions::Id